libr/debug/debug.c

```diff
@@ -1179,19 +1181,22 @@ R_API int r_debug_continue_kill(RDebug *dbg, int sig) {
 		}
 
 #if __linux__
-		if (reason == R_DEBUG_REASON_NEW_PID && dbg->follow_child) {
+		if (reason == R_DEBUG_REASON_NEW_PID) {
 #if DEBUGGER
-			/// if the plugin is not compiled link fails, so better do runtime linking
-			/// until this code gets fixed
-			static bool (*linux_attach_new_process) (RDebug *dbg, int pid) = NULL;
-			if (!linux_attach_new_process) {
-				linux_attach_new_process = r_lib_dl_sym (NULL, "linux_attach_new_process");
-			}
-			if (linux_attach_new_process) {
-				linux_attach_new_process (dbg, dbg->forked_pid);
+			if (dbg->follow_child) {
+				/// if the plugin is not compiled link fails, so better do runtime linking
+				/// until this code gets fixed
+				static bool (*linux_attach_new_process) (RDebug *dbg, int pid) = NULL;
+				if (!linux_attach_new_process) {
```

Comment:
this dlsym thing is a hack, and wont work on stripped builds of r2. This code can be inside an #if __linux__ instead of asuming the symbol is named like this and fail at runtime, the comment above must be removed and the code fixed